cmd/common/oci.go

```diff
@@ -256,12 +261,57 @@ func NewRunCommand(rootCmd *cobra.Command, runtime runtime.Runtime, hiddenColumn
 
 		timeoutDuration := time.Duration(timeoutSeconds) * time.Second
 
-		imageName := args[0]
+		var imageName string
+		if len(args) > 0 {
+			imageName = args[0]
+		}
+
+		paramValueMap := make(map[string]string)
+
+		if inFile != "" {
+			f, err := os.Open(inFile)
+			if err != nil {
+				return fmt.Errorf("opening gadget runtime configuration file %s: %w", inFile, err)
+			}
+			specs, err := gadgetconfig.InstanceSpecsFromReader(f)
+			f.Close()
+			if err != nil {
+				return fmt.Errorf("reading gadget runtime configuration file %s: %w", inFile, err)
+			}
+
+			detachedParam := runtimeParams.Get("detach")
+			isDetach := detachedParam != nil && detachedParam.AsBool()
+
+			if len(specs) > 1 && !isDetach {
+				return fmt.Errorf("multiple gadget runtime configurations found in file %s: this is only supported in combination with --detach", inFile)
+			}
+
+			if isDetach {
+				// Copy special oci params
+				ociParams.CopyToMap(paramValueMap, "operator.oci.")
```

Comment:
I don't understand this logic. It's copying some parameters to the map, but `ParamValues` on the spec is totally ignored. 

Perhaps something like can help:

```bash 
diff --git a/cmd/common/oci.go b/cmd/common/oci.go
index eb407ce0c..e428af186 100644
--- a/cmd/common/oci.go
+++ b/cmd/common/oci.go
@@ -288,8 +288,8 @@ func NewRunCommand(rootCmd *cobra.Command, runtime runtime.Runtime, hiddenColumn

                        if isDetach {
                                // Copy special oci params
-                               ociParams.CopyToMap(paramValueMap, "operator.oci.")
-                               return runInstanceSpecsDetached(ctx, runtime, specs, runtimeParams, paramValueMap,
+                               //ociParams.CopyToMap(paramValueMap, "operator.oci.")
+                               return runInstanceSpecsDetached(ctx, runtime, specs, runtimeParams, ociParams,
                                        gadgetcontext.WithDataOperators(ops...),
                                        gadgetcontext.WithTimeout(timeoutDuration),
                                        gadgetcontext.WithUseInstance(commandMode == CommandModeAttach),
@@ -387,11 +387,15 @@ func runInstanceSpecsDetached(
        runtime runtime.Runtime,
        specs []*gadgetconfig.InstanceSpec,
        runtimeParams *params.Params,
-       paramValueMap map[string]string,
+       ociParams *params.Params,
        runOptions ...gadgetcontext.Option,
 ) error {
        var merr error
        for _, spec := range specs {
+               // TODO: Clone map?
+               paramValueMap := spec.ParamValues
+               ociParams.CopyToMap(paramValueMap, "operator.oci.")
+
                imageName := spec.ImageName

                // Set some well-known params
```

Reply:
Yeah, I broke it with the recent push, sorry :/

Added some tests to make sure it's valid.